webui/react/src/pages/TaskList.settings.ts

```diff
@@ -0,0 +1,81 @@
+import { MINIMUM_PAGE_SIZE } from 'components/Table';
+import { BaseType, SettingsConfig } from 'hooks/useSettings';
+import { CommandState, CommandType } from 'types';
+
+export interface Settings {
+  search: string;
+  sortDesc: boolean;
+  sortKey: 'id' | 'name' | 'resourcePool' | 'startTime' | 'state' | 'type' | 'username';
+  state?: CommandState[];
+  tableLimit: number;
+  tableOffset: number;
+  type?: CommandType[];
+  user?: string[];
+}
+
+const config: SettingsConfig = {
+  settings: [
+    {
+      defaultValue: '',
+      isFilter: true,
```

Comment:
yeah maybe not the place to have it. It is convenient to have it as a property to have hook automatically handle it, but maybe it doesn't quite fit into the settings mold.

Reply:
I was gonna chat with you about this. maybe we still do later. if it's very convenient to have it maybe we do 🤔

Reply:
one thought I had around this is to enlarge the scope and maybe think about this as `useUserState`, where it will handles all the states that involve direct user interactions. It will then automatically handle behaviors around filtering, query string, internal state, local storage, and potentially more future hooks.